signer/src/transaction_coordinator.rs

```diff
@@ -799,6 +794,11 @@ where
                 "kind" => "complete-deposit"
             )
             .increment(1);
+
+            if &self.context.state().bitcoin_chain_tip() != chain_tip {
+                tracing::info!("new bitcoin chain tip, stopping coordinator activities");
+                return Ok(());
+            }
```

Comment:
I'm wondering if we should move the check at the beginning of the loop instead. Doing so we can remove the check from `construct_and_sign_stacks_response_transactions`.
If we want to keep them as is, we should probably add another check at the beginning of `construct_and_sign_stacks_response_transactions` to guard also the first deposit request.

Reply:
Moved in https://github.com/stacks-network/sbtc/pull/1518/commits/76605d37af16f8a0e1829173c6d47a83bb8174c3